src/wasm/wasm-emscripten.cpp

```diff
@@ -286,20 +290,42 @@ Function* EmscriptenGlueGenerator::generateAssignGOTEntriesFunction() {
     if (ex) {
       assert(ex->kind == ExternalKind::Function);
       f = wasm.getFunction(ex->value);
-    } else {
-      ImportInfo info(wasm);
-      f = info.getImportedFunction(ENV, g->base);
-      if (!f) {
-        Fatal() << "GOT.func entry with no import/export: " << g->base;
+      if (!sideModule) {
+        // This is exported, so must be one of the functions implemented here.
+        // Simply add it to the table, and use that index. The loader will
+        // know to reuse that index for other modules so they all share the
+        // same index and function pointer equality works.
+        // We may be able to do something for side modules as well, however,
+        // that would require at least updating the dylink section.
+        if (f->imported()) {
+          Fatal() << "GOT.func entry export but not implemented: " << g->base;
```

Comment:
This error could be better maybe?  if you hit this does it mean that GOT.func.foo is both imported *and* exported?

Reply:
Yes, that's equivalent. I'll rephrase the error that way.